packages/sdk/src/purchase/delegation.ts

```diff
@@ -0,0 +1,69 @@
+import { ethers, Signer, providers } from "ethers";
+
+import { IDelegationRegistry } from "../contracts/IDelegationRegistry";
+import { IDelegationRegistry__factory } from "../contracts/factories/IDelegationRegistry__factory";
+
+/**
+ * Define the registry for delegating a hot wallet to mint/claim airdrops on behalf of a vault wallet
+ * @see https://delegate.cash/
+ */
+export const DELEGATION_REGISTRY = "0x00000000000076a84fef008cdabe6409d2fe638b";
+
+/**
+ * @summary Error thrown when a user is a delegate on behalf of multiple vaults.
+ * @description Currently, only a single vault can be delegated to at a time.
+ * This error is thrown when a user has delegated to multiple vaults, and provides
+ * an easy way for SDK consumers to catch this error and display a helpful message
+ * to the user.
+ */
+export class MultipleDelegationsError extends Error {
+  constructor() {
+    super(
+      "Delegating multiple vaults is not currently supported. You may run into errors when minting allowlisted projects."
+    );
+    this.name = "MultipleDelegationsError";
+  }
+}
+
+/**
+ * @summary Get the delegation registry contract.
+ * @description This function is used to get the delegation registry contract
+ * so that we can call the getDelegationsByDelegate function to determine if
+ * the user has delegated to a vault.
+ * @param contractAddress the address of the delegation registry contract
+ * @param signerOrProvider ethers signer or provider
+ * @returns the delegation registry contract interface
+ */
+export const getDelegationRegistryContract = (
+  contractAddress: string,
+  signerOrProvider: Signer | providers.Provider
+): IDelegationRegistry => {
+  return IDelegationRegistry__factory.connect(
+    contractAddress,
+    signerOrProvider
+  );
+};
+
+/**
+ * @summary Get the vault that the user has delegated to.
```

Comment:
nit: I think some of the wording/naming is a little odd here - I would phrase this as: "Get the vault that the user is a delegate for". This is because the vault was the one that had to perform the delegation - the user/hot wallet is simply the delegate. 

for example, I would start description below with something like: "This function is used to determine if the user is the delegate of a vault."